btcec/schnorr/musig2/nonces.go

```diff
@@ -392,18 +373,71 @@ func AggregateNonces(pubNonces [][PubNonceSize]byte) ([PubNonceSize]byte, error)
 	if err != nil {
 		return finalNonce, err
 	}
+
 	combinedNonce2, err := combineNonces(func(n [PubNonceSize]byte) []byte {
 		return n[btcec.PubKeyBytesLenCompressed:]
 	})
 	if err != nil {
 		return finalNonce, err
 	}
 
-	copy(finalNonce[:], combinedNonce1.SerializeCompressed())
+	copy(finalNonce[:], JacobianToByteSlice(combinedNonce1))
 	copy(
 		finalNonce[btcec.PubKeyBytesLenCompressed:],
-		combinedNonce2.SerializeCompressed(),
+		JacobianToByteSlice(combinedNonce2),
 	)
 
 	return finalNonce, nil
 }
+
+// NonceToJacobian parses the nonce as a secp.Pubkey and converts the pubkey
+// to the passed result. If the nonce is a zero slice, the infinityPoint is
+// returned.
+func NonceToJacobian(nonce []byte, result *btcec.JacobianPoint) error {
+	if len(nonce) != 33 {
+		str := fmt.Sprintf("invalid nonce: invalid length: %x",
+			len(nonce))
+		return makeError(secp.ErrPubKeyInvalidLen, str)
+	}
+
+	format := nonce[0]
+	switch format {
+	case 0x00:
+		result.X.SetInt(0)
+		result.Y.SetInt(0)
+		result.Z.SetInt(0)
+		return nil
+	case secp.PubKeyFormatCompressedEven, secp.PubKeyFormatCompressedOdd:
+	default:
+		str := fmt.Sprintf("invalid nonce: unsupported format: %x",
+			format)
+		return makeError(secp.ErrPubKeyInvalidFormat, str)
+	}
+
+	noncePk, err := secp.ParsePubKey(nonce)
+	if err != nil {
+		return err
+	}
+	noncePk.AsJacobian(result)
+
+	return nil
+}
+
+// JacobianToByteSlice converts the passed secp.JacobianPoint to a secp.Pubkey
+// and serializes that to a byte slice. If the JacobianPoint is the infinity
+// point, a zero slice is returned.
+func JacobianToByteSlice(point secp.JacobianPoint) []byte {
+
+	if point.X == infinityPoint.X && point.Y == infinityPoint.Y {
+		return make([]byte, 33)
+	}
+
```

Comment:
You need to call `ToAffine()` before it's ok to just set the X and Y coords in the pubkey.